pkg/retry/retry.go

```diff
@@ -49,6 +53,8 @@ func IfNecessary(ctx context.Context, operation func() error, options *Options)
 	return err
 }
 
+var IsErrorRetryable = isRetryable
```

Comment:
Especially if `Options` already allows customizing the behavior, why do we need this global variable? One caller of `pkg/retry` setting this value would affect all other users in quite likely unexpected ways.

Just making the function public would work.